src/main/java/ticTacToe/game/GameFlow.java

```diff
@@ -0,0 +1,72 @@
+package ticTacToe.game;
+
+import ticTacToe.grid.Grid;
+import ticTacToe.grid.Rows;
+import ticTacToe.player.Player;
+import ticTacToe.player.PlayersFactory;
+import ticTacToe.ui.Ui;
+import java.util.HashMap;
+import java.util.Map;
+
+import static ticTacToe.game.Mark.*;
+import static ticTacToe.game.Mark.NOUGHT;
+
+public class GameFlow {
+
+    private final Ui ui;
+    private final Grid grid;
+    private final Rows rows;
+    private final PlayersFactory playersFactory;
+    private Player firstPlayer;
+    private Player secondPlayer;
+
+    public GameFlow(Ui ui, Grid grid, Rows rows) {
+        this.ui = ui;
+        this.grid = grid;
+        this.rows = rows;
+        this.playersFactory = new PlayersFactory();
+        ui.welcomePlayer();
+    }
+
+    public void runGame() {
+        firstPlayer = playersFactory.firstPlayer(ui);
+        String currentMark = ui.askForStarter();
+        secondPlayer = playersFactory.secondPlayer(firstPlayer.getMark());
+        gameFlow(currentMark);
+    }
+
+    private void gameFlow(String currentMark) {
+        while (!grid.allOccupiedCells()) {
+            String positionChosen = currentPlayer(currentMark).makeMove(ui, grid, rows);
+            grid.addMark(currentMark, positionChosen);
+            if (rows.isWinning(grid.getSize())) {
+                ui.declareWinner(rows.winningMark(grid.getSize()));
+            } else {
+                currentMark = switchPlayerMark(currentMark);
+                gameFlow(currentMark);
+            }
+            break;
```

Comment:
Can you re-work the conditions in the while statement so that you don't need the break statement?

Reply:
In next PR - stay tuned! 🤓